Server/service/networkService.js

```diff
@@ -110,6 +121,55 @@ class NetworkService {
 		}
 	}
 
+	async handleHardwareUpdate(monitor, hardwareData) {
+		//Get Thresholds
+		const thresholds = monitor?.thresholds;
+		if (thresholds === undefined) {
+			return;
+		}
+		//Get Values
+		const cpuUsage = hardwareData?.cpu?.usage_percent;
+		const memoryUsage = hardwareData?.memory?.usage_percent;
+		const diskUsage = hardwareData?.disk[0]?.usage_percent;
+
+		// Return early if no values
+		if (cpuUsage === undefined && memoryUsage === undefined && diskUsage === undefined) {
+			return;
+		}
+
+		// Return early if all values are below thresholds
+		if (
+			cpuUsage < thresholds.usage_cpu &&
+			memoryUsage < thresholds.usage_memory &&
+			diskUsage < thresholds.usage_disk
+		) {
+			return;
+		}
+
+		let context = {
+			message: "Threshold(s) exceeded:",
+		};
+		if (cpuUsage > thresholds.usage_cpu) {
+			context.cpu = `CPU USAGE: ${cpuUsage * 100}% > ${thresholds.usage_cpu * 100}%`;
+		}
+
+		if (memoryUsage > thresholds.usage_memory) {
+			context.memory = `MEMORY USAGE: ${memoryUsage * 100}% > ${thresholds.usage_memory * 100}%`;
+		}
+
+		if (diskUsage > thresholds.usage_disk) {
+			context.disk = `DISK USAGE: ${diskUsage * 100}% > ${thresholds.usage_disk * 100}%`;
+		}
+
+		const config = {
+			monitor: monitor,
+			template: "thresholdViolatedTemplate",
+			context: context,
+			subject: `Threshold Violated for ${monitor.name}`,
+		};
+		this.handleNotification(config);
```

Comment:
Do we need await here?

Reply:
We don't need to await here as the result of `handleNotification` isn't needed for anything, we just want to fire off the notification and then keep executing code.  If the notification fails for whatever reason we want to keep executing